### PR TITLE
Setup coverage reporting

### PR DESCRIPTION
**Pull request by blueyed**

### CODE REVIEW

**Comment by blueyed** [comment]
Not yet.. ;)
Can this be squash-merged then here?  (otherwise I'll force-push once it is ok)

File: .coveragerc [hunk 1]
```diff
@@ -0,0 +1,16 @@
+[run]
+include =
+  pluggy/*
+  testing/*
+  */lib/python*/site-packages/pluggy/*
+  */pypy*/site-packages/pluggy/*
+  *\Lib\site-packages\pluggy\*
+  *\Lib\site-packages\pytest.py
```
Comment (blueyed) [inline .coveragerc]: will remove that line.

**Comment by blueyed** [comment]
Report looks good now: https://codecov.io/gh/pytest-dev/pluggy/pull/197/tree

**Comment by goodboy** [comment]
@blueyed I usually `git rebase -i` and squash and then force push the new history.

Also probably want @nicoddemus to have a look since he's the CI expert then round these parts.

**Comment by blueyed** [comment]
Well, @nicoddemus does not like squash-merging too much.. ;)
Squashed.
The downside is a new CI run.. :/

**Comment by goodboy** [comment]
>Report looks good now

@blueyed pretty interesting to see how much we're missing actually. Though I'm not totally surprised.
Doesn't seem like it's showing which lines aren't getting hit though - that because we don't have a baseline I guess. Sorry total newb with `codecov.io`.

**Comment by blueyed** [comment]
Yeah, no baseline.

> Doesn't seem like it's showing which lines aren't getting hit though

the PR  view is buggy without a base I assume.
You can browse the commit: https://codecov.io/gh/pytest-dev/pluggy/tree/d8896f94757084bf83ed6e6ceb5b02928d84cfb2

**Comment by blueyed** [comment]
It also needs eager coverage setup (with pytest-cov), or use coverage.py by itself, since pytest imports it already apparently.

**Comment by blueyed** [comment]
92% looks better.. :)

**Comment by nicoddemus** [comment]
> Well, @nicoddemus does not like squash-merging too much.. ;)

Actually I don't mind, you are confusing me with someone else. 😁

**Comment by nicoddemus** [comment]
My previous merge caused this one to get into conflict. I've fixed the conflicts myself, we're good to go once it passes again. 👍 

@blueyed would appreciate if you double check by conflict resolution. 😁

**Comment by blueyed** [comment]
@nicoddemus 
LGTM, cannot merge it myself.

**Pull request merged**